internal/namespaces/instance/v1/custom_server.go

```diff
@@ -169,6 +169,105 @@ func bootscriptMarshalerFunc(i interface{}, opt *human.MarshalOpt) (string, erro
 	return bootscript.Title, nil
 }
 
+//
+// Builders
+//
+
+func serverUpdateBuilder(c *core.Command) *core.Command {
+	type instanceUpdateServerRequestCustom struct {
+		*instance.UpdateServerRequest
+		IP *instance.NullableStringValue
+	}
+
+	IPArgSpec := &core.ArgSpec{
+		Name:  "ip",
+		Short: `IP that should be attached to the server (use ip=none to remove)`,
+	}
+
+	c.ArgsType = reflect.TypeOf(instanceUpdateServerRequestCustom{})
+
+	c.ArgSpecs = append(c.ArgSpecs, IPArgSpec)
+
+	c.Run = func(ctx context.Context, argsI interface{}) (i interface{}, e error) {
+		customRequest := argsI.(*instanceUpdateServerRequestCustom)
+
+		updateServerRequest := customRequest.UpdateServerRequest
+
+		attachIPRequest := (*instance.UpdateIPRequest)(nil)
+
+		detachIP := false
+
+		client := core.ExtractClient(ctx)
+		api := instance.NewAPI(client)
+
+		getServerResponse, err := api.GetServer(&instance.GetServerRequest{
+			Zone:     "",
+			ServerID: customRequest.ServerID,
+		})
+		if err != nil {
+			return "", err
+		}
+
+		switch {
+		case customRequest.IP == nil:
+			// ip is not set
+			// do nothing
+
+		case customRequest.IP.Null:
+			// ip=none
+			// remove server from ip, only if a Server and an IP are linked.
+			if getServerResponse.Server.PublicIP != nil {
+				detachIP = true
+			}
+
+		default:
+			// ip=<anything>
+			// update ip
+			if getServerResponse.Server.PublicIP != nil {
+				detachIP = true
+			}
+			attachIPRequest = &instance.UpdateIPRequest{
+				IP: customRequest.IP.Value,
+				Server: &instance.NullableStringValue{
+					Value: customRequest.ServerID,
+				},
+			}
+		}
+
+		// Instance API does not support detaching the existing IP and then attaching a new one to the same server
+		// in 1 call only.
+		// We need to do it manually in 2 calls.
```

Comment:
We have to test only `api.UpdateIP(attachIPRequest)`, IIRC it will automatically detach the old IP. 🤔

Reply:
We need to do it in 2 calls. 

```
> scw instance ip list
ID                                    ADDRESS         REVERSE  ORGANIZATION                          SERVER ID                             SERVER NAME
95c94531-cba7-4e57-aa5e-b6a4dd7a816c  163.172.170.53           aba2d0d0-b01d-4d88-b322-935edc96d0fd
cd09e94b-64b9-4c6f-a9b0-e1ac4c7d793e  51.158.64.183            aba2d0d0-b01d-4d88-b322-935edc96d0fd  aa4fedfc-4937-4ce9-8e05-b5f8b2b5678c  scw-dreamy-saha
9e9d41c0-b586-4f46-bbcb-06271fe770f7  51.15.247.243            aba2d0d0-b01d-4d88-b322-935edc96d0fd
```

```
> curl -H "X-Auth-Token: $SCW_API_TOKEN" -H 'Content-Type: application/json' -X PATCH https://api.scaleway.com/instance/v1/zones/fr-par-1/ips/<ip> --data '{"server": "<uuid>"}'
{"fields": {"server": ["Server '<uuid>' is already attached to an IP address."]}, "message": "Server '<uuid>' is already attached to an IP address.", "type": "invalid_request_error"}
```